src/components/expandableSection/expandableSection.tsx

```diff
@@ -127,15 +164,91 @@ export const ExpandableSection: FC<ExpandableSectionProps> = (props) => {
     if (onToggle) onToggle(newIsOpen);
   };
 
+  const handleActionClick = (action: ExpandableSectionAction) => {
+    action.onClick();
+  };
+
   const titleElement = typeof title === 'string' ? <StyledTitleDiv>{title}</StyledTitleDiv> : title;
-  const iconElement = (
+  const iconElement = iconName ? (
     <StyledCaretIconDiv isOpen={isOpen}>
       <Icon name={iconName} size={16} />
     </StyledCaretIconDiv>
-  );
+  ) : null;
+
+  // If no actions, render the section normally
+  if (actions.length === 0)
```

Comment:
Instead of doing this, could we return null in the `renderActions` method?
```typescript
const renderActions = () => {
    if (actions.length === 0) return null;

    if (groupActions) {
      // Group all actions into a dropdown menu
      return (
```